src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts

```diff
@@ -43,12 +43,20 @@ export default function useClientSideSmartOrderRouterTrade<TTradeType extends Tr
   useEffect(() => {
     async function fetchQuote() {
       if (queryArgs) {
```

Comment:
Hm if query args isn’t defined you may never reset to loading(false)?might be better to set loading right before the await get client side quote ?

Reply:
or wrap the whole thing in a `try {} finally {}`